google/cloud/spanner_dbapi/connection.py

```diff
@@ -235,6 +235,7 @@ def close(self):
         ):
             self._transaction.rollback()
 
+        self.database._pool.clear()
```

Comment:
If a user has created multiple connections for the same database, then I think this will invalidate the session pool of all those connections, while he/she is only closing this connection.

Reply:
Yeah, I had a thought about that. To achieve this one will need to create a pool manually and propagate it to several `connect()` calls. Is it a correct way of use? I mean, what prevents user from trying to propagate one pool (binded to a single database) onto connections to several different databases.

Every `Database` in Python initiates a pool for itself (if non given), which makes me thing it's not intended to be used with global pools.

Reply:
Yeah, I should have looked a little bit more in detail into the `connection.py` file. The current implementation requires a user to pass in a `database` when a connection is created. That basically means that we are asking the user of the library to initialize and maintain a database reference themselves. That again means that the `Connection` class should not open or close any session pools for the `database`. Instead, that is the responsibility of the caller that has already obtained a reference to the `database`.